packages/story-editor/src/app/highlights/provider.tsx

```diff
@@ -17,52 +17,62 @@
  * External dependencies
  */
 import { useCallback, useMemo, useState } from '@googleforcreators/react';
-import PropTypes from 'prop-types';
-
+import type { PropsWithChildren } from 'react';
 /**
  * Internal dependencies
  */
-import { useStory } from '../story';
+import { useStory } from '../story/useStory';
+import type {
+  setHighlightProps,
+  selectElementProps,
+  HighlightType,
+} from '../../types/highlightsProvider';
 import Context from './context';
 import { STATES } from './states';
 
-function HighlightsProvider({ children }) {
-  const [highlighted, setHighlighted] = useState({});
-  const { setSelectedElementsById, setCurrentPage } = useStory(
-    ({ actions }) => ({
-      setSelectedElementsById: actions.setSelectedElementsById,
-      setCurrentPage: actions.setCurrentPage,
-    })
-  );
+type HighlightState = {
+  [k in HighlightType]: {
+    focus: boolean;
+    showEffect?: boolean;
+    tab: string;
+    section: string;
+  };
+};
+
+function HighlightsProvider({ children }: PropsWithChildren<unknown>) {
+  const [highlighted, setHighlighted] = useState<
+    HighlightState | Record<never, never>
+  >({});
```

Comment:
`Record<never, never>` is for the empty object? If you make `[k in HighlightType]:` above optional, it should work without it as well probably, no?